src/lib/hidden_node.ts

```diff
@@ -0,0 +1,31 @@
+import {Context} from './context';
+
+/**
+ * Condition for applying rule 2A
+ * @param {HTMLElement} elem - The element whose accessible name is being calculated
+ * @param {Context} context - Additional information relevant to the name computation for elem
+ * @return {boolean} - Whether of not elem satisfies the condition for rule 2A
+ */
+function hiddenNodeCondition(elem: HTMLElement, context: Context): boolean {
```

Comment:
I don't think that "hiddenNodeCondition" is a very descriptive name here. This file seems to abstract step 2A, so we should probably name it "step2ACondition" or something similar. 

The fact that a node is "hidden" is a separate concern that is only part of step 2A. Even if a "hidden" node is referenced by aria-labelledby, it should still be considered "hidden".